packages/client-cli/lib/get-type.mjs

```diff
@@ -70,22 +70,24 @@ export function getType (typeDef, methodType, spec) {
     }).join(' | ')
   }
   if (typeDef.type === 'object') {
-    if (!typeDef.properties || Object.keys(typeDef.properties).length === 0) {
+    const additionalPropsObj = typeDef?.additionalProperties?.properties
+    const objProperties = typeDef.properties || additionalPropsObj
+    if (!objProperties || Object.keys(objProperties).length === 0) {
       // Object without properties
       return 'object'
     }
-    let output = '{ '
+    let output = additionalPropsObj ? 'Record<string, { ' : '{ '
```

Comment:
The condition for generating a `Record` should only be that `typeDef?.additionalProperties` has type `object` instead of `boolean`. Then that object is the JSON schema that should be translated in the type on the second parameter to `Record`. This JSON schema can actually be anything, not necessarily another object.

For example, given this JSON schema:

```json
{
  "type": "object",
  "additionalProperties": { "type": "number" }
}
```

The generated type should be:

```ts
Record<string, number>
```

So in order to do this, after generating `Record<, `, we should recurse and call `getType(typeDef?.additionalProperties)`.

---

Note though that the JSON schema, in addition to `additionalProperties`, could have `properties` and `requiredProperties` too, so they all should be `&`-ed together.

For example, expanding on the previous JSON schema:

```json
{
  "type": "object",
  "properties": { "foo": { "type": "string" } },
  "additionalProperties": { "type": "number" }
}
```

This is what `openapi-ts` generates:

```ts
type T = {
    foo?: string;
} & {
    [key: string]: number;
}
```

(Which, to be fair, could also be improved. There is no way to populate `foo` here in a compliant way —any value either won't be a `string` or a `number`, so it will fail to typecheck.)

Reply:
[8ef4066](https://github.com/platformatic/platformatic/pull/3405/commits/8ef4066dbfb42683439555b91ba54b9549888d7f)